src/test/groovy/com/palantir/gradle/versions/VersionsPropsTest.java

```diff
@@ -59,4 +59,60 @@ void fails_to_load_illegal_version() throws IOException {
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessageContaining("invalid constraint");
     }
+
+    @Test
+    void ignores_comment_on_its_own_line() throws IOException {
+        Path propsFile = tempDir.resolve("versions.props");
+        Files.writeString(
+                propsFile, "# comment on its own line\ncom.palantir.test:test = 1.0.0", StandardCharsets.UTF_8);
+
+        VersionsProps versionsProps = VersionsProps.loadFromFile(propsFile);
+        assertThat(versionsProps.getFuzzyResolver().exactMatches()).containsExactly("com.palantir.test:test");
+        assertThat(versionsProps.getFuzzyResolver().globs()).isEmpty();
+    }
+
+    @Test
+    void ignores_comment_on_same_line() throws IOException {
+        Path propsFile = tempDir.resolve("versions.props");
+        Files.writeString(propsFile, "com.palantir.test:test = 1.0.0  # comment on same line", StandardCharsets.UTF_8);
+
+        VersionsProps versionsProps = VersionsProps.loadFromFile(propsFile);
+        assertThat(versionsProps.getFuzzyResolver().exactMatches()).containsExactly("com.palantir.test:test");
+        assertThat(versionsProps.getFuzzyResolver().globs()).isEmpty();
+    }
+
+    @Test
+    void ignores_comment_on_same_line_with_no_hash() throws IOException {
+        Path propsFile = tempDir.resolve("versions.props");
+        Files.writeString(
+                propsFile, "com.palantir.test:test = 1.0.0  comment on same line with no hash", StandardCharsets.UTF_8);
```

Comment:
this seems less than ideal, but I'm not sure we gain much by making this invalid